### PR TITLE
IGNITE-14424

### PR DESCRIPTION
**Pull request by xtern**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

**Review by NSAmelchev — APPROVED** [review]

File: modules/core/src/main/java/org/apache/ignite/internal/managers/encryption/GridEncryptionManager.java [hunk 1]
```diff
@@ -1113,6 +1113,18 @@ private void releaseWalKeys(long segmentIdx) {
         }
     }
 
+    /** {@inheritDoc} */
+    @Override public void onDoneAfterTopologyUnlock(GridDhtPartitionsExchangeFuture fut) {
+        if (fut.activateCluster() || fut.localJoinExchange()) {
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/internal/managers/encryption/GridEncryptionManager.java:1118]: What if the current node joins an inactive cluster?

File: modules/core/src/test/java/org/apache/ignite/internal/encryption/CacheGroupReencryptionTest.java [hunk 2]
```diff
@@ -631,6 +631,38 @@ public void testReencryptionOnUnstableTopology() throws Exception {
...
+        createEncryptedCache(node0, node1, cacheName(), null);
+
+        loadData(100_000);
+
+        node0.encryption().changeCacheGroupKey(Collections.singleton(cacheName())).get();
+
+        assertTrue(isReencryptionInProgress(Collections.singleton(cacheName())));
+
+        node0.cluster().state(ClusterState.INACTIVE);
+
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/internal/encryption/CacheGroupReencryptionTest.java:655]:
```
        stopGrid(GRID_1);
        startGrid(GRID_1);
```
Reproduces the problem I've mentioned above

File: modules/core/src/main/java/org/apache/ignite/internal/managers/encryption/GridEncryptionManager.java (hunk 1 above)
Comment (xtern) [reply modules/core/src/main/java/org/apache/ignite/internal/managers/encryption/GridEncryptionManager.java:1118]: If a node joins an inactive cluster, no PME occurs.

File: modules/core/src/test/java/org/apache/ignite/internal/encryption/CacheGroupReencryptionTest.java (hunk 2 above)
Comment (xtern) [reply modules/core/src/test/java/org/apache/ignite/internal/encryption/CacheGroupReencryptionTest.java:658]: Added this case and fix for it.
(from my point of view, the name of the ``isReencryptionInProgress`` method in the test is not entirely correct - this method checks that re-encryption is required, but it doesn’t say whether the re-encryption task is running (), I can’t remember why this name was chosen)

**Review by alex-plekhanov — APPROVED** [review]

**Pull request merged**